vllm/distributed/communication_op.py

```diff
@@ -24,10 +25,18 @@ def graph_capture_mode():
     #
     # Note that custom allreduce will have a runtime check, if the tensor size
     # is too large, it will fallback to the next available option.
+    # In summary: When using CUDA graph, we use
+    # either custom all-reduce kernel or pynccl. When not using CUDA
+    # graph, we use either custom all-reduce kernel or PyTorch NCCL.
+    # We always prioritize using custom all-reduce kernel but fall back
+    # to PyTorch or pynccl if it is disabled or not supported.
     pynccl_comm = get_tp_pynccl_communicator()
+    ca_comm = get_tp_ca_communicator()
     assert pynccl_comm is not None
+    assert ca_comm is not None
     with pynccl_comm.change_state(enable=True,
-                                  stream=torch.cuda.current_stream()):
+                                  stream=torch.cuda.current_stream()), \
+                                  ca_comm.capture():
```

Comment:
nit:
```suggestion
    with (pynccl_comm.change_state(enable=True,
                                  stream=torch.cuda.current_stream()), 
                                  ca_comm.capture()):
```

Reply:
fixed in [105d14c](https://github.com/vllm-project/vllm/pull/4754/commits/105d14cfe97924f762771d02f98299b07e144cf7).